src/driver/result.rs

```diff
@@ -281,7 +281,11 @@ pub unsafe fn malloc_async(
     num_bytes: usize,
 ) -> Result<sys::CUdeviceptr, DriverError> {
     let mut dev_ptr = MaybeUninit::uninit();
+    #[cfg(feature = "alloc_async")]
     sys::cuMemAllocAsync(dev_ptr.as_mut_ptr(), num_bytes, stream).result()?;
+
+    #[cfg(not(feature = "alloc_async"))]
+    sys::cuMemAlloc_v2(dev_ptr.as_mut_ptr(), num_bytes).result()?;
```

Comment:
Let's change the feature name to `sync_alloc`, reverse and remove it from the default features list. Would prefer opt-in to the non-standard behavior, rather than opting out of it.

```suggestion
    #[cfg(not(feature = "sync_alloc"))]
    sys::cuMemAllocAsync(dev_ptr.as_mut_ptr(), num_bytes, stream).result()?;

    #[cfg(feature = "sync_alloc")]
    sys::cuMemAlloc_v2(dev_ptr.as_mut_ptr(), num_bytes).result()?;
```

Reply:
That's a feature that removes something, no ?

I'll take it if you want to include in the lib.

Actually thinking about it, what about, not having a feature flag, and having 2 functions (pushing the burden on the user to figure it out, which IMO is better here, since it should be my choice to figure out which function to calll)

Reply:
Yeah that could work - we'd have to figure out how to integrate into CudaDevice allocation functions I think

Reply:
Probably.